sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala

```diff
@@ -476,6 +479,18 @@ private object CosmosWriteConfig {
     helpMessage = s"Cosmos DB Item Write Max Pending Operations." +
       s" If not specified it will be determined based on the Spark executor VM Size")
 
+  private val bulkMaxConcurrentPartitions = CosmosConfigEntry[Int](
+    key = CosmosConfigNames.WriteBulkMaxConcurrentPartitions,
+    mandatory = false,
+    parseFromStringFunction = bulkMaxConcurrencyAsString => bulkMaxConcurrencyAsString.toInt,
+    helpMessage = s"Cosmos DB Item Write Max Concurrent Cosmos Partitions." +
```

Comment:
Should we include some guidance here of the sort: If you are repartitioning to X, make sure to use this to match that X value? I wonder between this setting and `bulkmaxPendingOperations` it sounds all a bit obscure and complex?

Reply:
Yes - I will expand on it in the Bulk Ingestion guide. And I have intentionally not documented this new config in the config reference yet. Ultimately both settings should never be needed (and I'll explain that in the guide) - they are more of an emergency knob if something goes wrong.